source/megengine.js

```diff
@@ -3,30 +3,96 @@
 
 var megengine = megengine || {};
 var base = base || require('./base');
+var zip = zip || require('./zip');
 
 megengine.ModelFactory = class {
 
     match(context) {
+        megengine.modelType = '';
+        const identifier = context.identifier;
+        const extension = identifier.split('.').pop().toLowerCase();
+        const stream = context.stream;
+        if (extension === 'mge' && stream.length >= 8) {
+            const buffer = stream.peek(Math.min(32, stream.length));
+            if (buffer.length >= 12) {
+                const magic = String.fromCharCode(...buffer.slice(8, 12));
+                if (magic === 'mgv2') {
+                    megengine.modelType = 'megengine.flatbuffers';
+                    return megengine.modelType;
+                }
+                throw new megengine.Error('Only support megengine c++ model v2');
+            }
+        }
         const obj = context.open('pkl');
         if (obj.__class__ && obj.__class__.__module__ === 'megengine.traced_module.traced_module' && obj.__class__.__name__ === 'TracedModule') {
-            return 'megengine.pickle';
+            megengine.modelType = 'megengine.pickle';
         }
-        return '';
+        return megengine.modelType;
     }
 
     open(context) {
         return context.metadata('megengine-metadata.json').then((metadata) => {
-            const obj = context.open('pkl');
-            return new megengine.Model(metadata, obj);
+            switch (megengine.modelType) {
+                case 'megengine.pickle': {
+                    const obj = context.open('pkl');
+                    return new megengine.Model(metadata, obj);
+                }
+                case 'megengine.flatbuffers': {
+                    return context.require('./megengine-schema').then(() => {
+                        megengine.schema_main = flatbuffers.get('megengine').mgb.serialization.fbs;
+                        megengine.schema_v2 = megengine.schema_main.v2;
+                        megengine.schema_param = megengine.schema_main.param;
+                        megengine.schema_dtype = megengine.schema_main;
+                        let model = null;
+                        const attachments = new Map();
+
+                        const stream = context.stream;
+                        stream.skip(4);
+                        try {
+                            const reader = flatbuffers.BinaryReader.open(stream);
+                            model = megengine.schema_v2.Model.create(reader);
+                        } catch (error) {
+                            const message = error && error.message ? error.message : error.toString();
+                            throw new megengine.Error('File format is not megengine.Model (' + message.replace(/\.$/, '') + ').');
+                        }
+                        try {
+                            const archive = zip.Archive.open(stream);
```

Comment:
Remove unused code and dependencies. For example, `zip` module is unused and the code seems to be copied from another format?

Reply:
done